### PR TITLE
[FIXED JENKINS-42017] Delegate jstack invocation to dumpling

### PR DESCRIPTION
**Pull request by olivergondza**
The particular problem is addressed in https://github.com/olivergondza/dumpling/blob/master/core/src/main/java/com/github/olivergondza/dumpling/factory/PidRuntimeFactory.java#L67-L77.

### CODE REVIEW

File: src/main/java/org/jenkinsci/test/acceptance/controller/LocalController.java [hunk 1]
```diff
@@ -374,12 +375,12 @@ private void diagnoseFailedLoad(Exception cause) throws IOException {
...
-                if (jstack.waitFor() == 0) {
-                    StringWriter writer = new StringWriter();
-                    IOUtils.copy(jstack.getInputStream(), writer);
-                    return writer.toString();
-                }
+                ThreadDumpRuntime runtime = new PidRuntimeFactory().fromProcess(pid);
+                ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                PrintStream printStream = new PrintStream(baos);
+                runtime.toString(printStream, ModelObject.Mode.MACHINE);
+                printStream.close();
```
Comment (scoheb) [inline src/main/java/org/jenkinsci/test/acceptance/controller/LocalController.java]: Do we need to put this in a finally block?

Comment (olivergondza) [inline src/main/java/org/jenkinsci/test/acceptance/controller/LocalController.java]: The only point for that stream to exist is to collect the output. `close()` effectively flushes the writer to ensure whole content is written to `toString()` returns everything. If exception gets propagated we do no longer care for the output so flushing is not necessary.

**Comment by mafraba** [comment]
looks good!

**Comment by batmat** [comment]
@olivergondza I'm releasing, please shout if you disagree? Thanks!

**Comment by batmat** [comment]
1.51 is out

**Pull request merged**